### ISSUE TITLE
Support multiple watchers

### ISSUE THREAD

**Issue by davidbrochart** [willmcgugan/declare#7]
Currently, it is posbbile to set [only one watcher](https://github.com/willmcgugan/declare/blob/89cdc69417f25dca2484a09a65b9e560aaa52898/src/declare/_declare.py#L85).
It would be great if multiple watchers were supported:
```py
from declare import Declare


class Foo:
    val = Declare[int](0)

    @val.watch
    def _watch_val1(self, old: int, new: int) -> None:
        print(f"val changed from {old} to {new}.")

    @val.watch
    def _watch_val2(self, old: int, new: int) -> None:
        print(f"yes, val changed from {old} to {new}!")
```

**Comment by davidbrochart** [comment]
FWIW I worked around that limitation with a [NestedWatchDecorator](https://github.com/davidbrochart/ypywidgets/blob/e7a5349c4628e985ebf068c0a89a0adc88a8d211/ypywidgets/declare.py#L12-L24).

**Comment by willmcgugan** [comment]
That could be made to work.

Although I'd be a little wary it might be tricky to debug if you have more than one watcher. It could be easy to lose track of what methods are watchers.

Did you have a particular use-case in mind?

**Comment by davidbrochart** [comment]
Yes, it's related to #11 where watchers are created from outside the original class.

**Comment by davidbrochart** [comment]
I opened #12.
